src/App.tsx

```diff
@@ -8,7 +8,66 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+
+import { getTodos } from './api';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState<number>(0);
+  const [status, setStatus] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  const selectStatus = (value: string) => {
+    setStatus(value);
+  };
+
+  const setFilterQuery = (value: string) => {
+    setQuery(value);
+  };
+
+  const selectTodoId = (todoId: number) => {
+    setSelectedTodoId(todoId);
+  };
+
+  const closeTodoModal = () => {
+    setSelectedTodoId(0);
+  };
+
+  const visibleTodos = todos.filter(todo => {
+    const isSearchQueryMatch = todo.title.toLowerCase().includes(query
+      .toLowerCase()
+      .split(' ')
+      .filter(Boolean)
+      .join(' '));
+
+    let isStatusMatch = true;
+
+    switch (status) {
+      case 'active':
+        isStatusMatch = !todo.completed;
+        break;
+
+      case 'completed':
+        isStatusMatch = todo.completed;
+        break;
+
+      default:
+        isStatusMatch = true;
+    }
+
+    return isSearchQueryMatch && isStatusMatch;
+  });
```

Comment:
it would be better to move filter logic to separate helper function
```suggestion
  const visibleTodos = useMemo(() => (
  	getFilteredTodos(todos, query, statusFilter
  	), [todos, query, statusFilter])
```

Reply:
and wrap it on useMemo